src/Stratis.Bitcoin.IntegrationTests/WalletTests.cs

```diff
@@ -92,6 +94,81 @@ public void WalletCanReceiveAndSendCorrectly()
+                Key key = wallet.GetExtendedPrivateKeyForAddress("123456", addr).PrivateKey;
+
+                stratisSender.SetDummyMinerSecret(new BitcoinSecret(key, stratisSender.FullNode.Network));
+                int maturity = (int)stratisSender.FullNode.Network.Consensus.Option<PowConsensusOptions>().CoinbaseMaturity;
+                stratisSender.GenerateStratis(maturity + 51);
+
+                // Wait for block repo for block sync to work.
+                TestHelper.WaitLoop(() => TestHelper.IsNodeSynced(stratisSender));
+                
+                Assert.Equal(Money.COIN * 150 * 50, stratisSender.FullNode.WalletManager().GetSpendableTransactionsInWallet("mywallet").Sum(s => s.Transaction.Amount));
+
+                // Sync both nodes.
+                stratisSender.CreateRPCClient().AddNode(stratisReceiver.Endpoint, true);
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(stratisReceiver, stratisSender));
+
+                // Get 50 unused addresses from the receiver.
+                IEnumerable<HdAddress> recevierAddresses = stratisReceiver.FullNode.WalletManager()
+                    .GetUnusedAddresses(new WalletAccountReference("mywallet", "account 0"), 50);
+
+                List<Recipient> recipients = recevierAddresses.Select(address => new Recipient
+                    {
+                        ScriptPubKey = address.ScriptPubKey,
+                        Amount = Money.COIN
+                    })
+                    .ToList();
+
+                var transactionBuildContext = new TransactionBuildContext(
+                    new WalletAccountReference("mywallet", "account 0"), recipients, "123456")
+                    {
+                        FeeType = FeeType.Medium,
+                        MinConfirmations = 101
+                    };
+
+                Transaction transaction = stratisSender.FullNode.WalletTransactionHandler().BuildTransaction(transactionBuildContext);
+
+                // Broadcast to the other node.
+                stratisSender.FullNode.NodeService<WalletController>().SendTransaction(new SendTransactionRequest(transaction.ToHex()));
+
+                // Wait for the trx's to arrive.
+                TestHelper.WaitLoop(() => stratisReceiver.CreateRPCClient().GetRawMempool().Length > 0);
+                TestHelper.WaitLoop(() => stratisReceiver.FullNode.WalletManager().GetSpendableTransactionsInWallet("mywallet").Any());
+
+                Assert.Equal(Money.COIN * 50, stratisReceiver.FullNode.WalletManager().GetSpendableTransactionsInWallet("mywallet").Sum(s => s.Transaction.Amount));
+                Assert.Null(stratisReceiver.FullNode.WalletManager().GetSpendableTransactionsInWallet("mywallet").First().Transaction.BlockHeight);  
+
+                // Generate two new blocks do the trx is confirmed.
+                stratisSender.GenerateStratis(1, new List<Transaction> { transaction.Clone() });
```

Comment:
dont use this method use instead `GenerateStratisWithMiner` this will already take the trx in mempool and include it in a block.

Reply:
Can you add a [deprecated] attribute on the method `GenerateStratis`

Reply:
It's used in quite a few places.  

![image](https://user-images.githubusercontent.com/16000683/37916997-3d0b52da-3115-11e8-8bf0-63a914c3814f.png)